packages/api/server/ws.mts

```diff
@@ -237,8 +237,30 @@ async function depsInstall(payload: DepsInstallPayloadType) {
           source: updatedJsonSource,
           status: 'idle',
         };
-        updateSession(session, { cells: replaceCell(session, updatedCell) }, false);
-        wss.broadcast(`session:${session.id}`, 'cell:updated', { cell: updatedCell });
+
+        const updatedSession = await updateSession(
+          session,
+          { cells: replaceCell(session, updatedCell) },
+          false,
+        );
+
+        wss.broadcast(`session:${updatedSession.id}`, 'cell:updated', { cell: updatedCell });
+
+        if (updatedSession.metadata.language === 'typescript') {
+          const mustCreateTsServer = !tsservers.has(updatedSession.id);
+
+          // Make sure to handle the following case here:
+          //
+          // 1. User creates a new typescript Srcbook
+          // 3. There is no tsserver running because it relies on the typescript package in the Srcbook's node modules, which are not yet installed.
```

Comment:
> Are we handling 3 ?

Yes, that's what this code is doing. Line 258 creates a new tsserver process if there is not one running. At this point, it should work fine because this code runs after deps are installed.